src/dosext/mfs/mfs.c

```diff
@@ -1367,7 +1373,9 @@ static struct dir_list *get_dir(char *name, char *mname, char *mext, int drive)
     entry->attr = get_dos_attr(NULL,entry->mode,entry->hidden);
 
     dos_closedir(cur_dir);
-    return (dir_list);
+    if (!dir_list && patherror)
+      *patherror = 0;
```

Comment:
I guess you can set it to 0 initially once,
to not duplicate things.

Reply:
Yes